### PR TITLE
Follow pagination to get all records on all pages

### LINKED ISSUE #10: Follow the pagination to make sure we scrape all results
Right now we're only scraping the first page of results, we need to follow the "Next" link in the pagination to make sure we're getting everything

### PR DESCRIPTION
**Pull request by aldrio**
This closes #10 

This was not as straightforward as following a simple url in the "Next" button's <a> tag. ASP.NET handles a page like this, which is a sort of "proto-SPA", using something called "postbacks" for interacting with forms and buttons on the page.

So I've added a couple of methods to our spider for interacting with postbacks (which the initial form submission also was one). The next button's href is actually a javascript snippet that performs a postback, so for our purposes I've added `follow_postback_link` to parse the postback info and follow it.

### CODE REVIEW

File: src/demolition_spider.py [hunk 1]
```diff
@@ -34,22 +33,57 @@ def __init__(
...
+        self,
+        response: Response,
+        href: str = "",
+        **kwargs,
+    ):
+        """
+        Parses and follows an ASP.NET postback link
+        """
+
+        m = re.search(r"__doPostBack\('(.+)'\s*,\s*'(.*)'\)", href)
```
Comment (thashepherd) [inline src/demolition_spider.py:69]: Nice

File: src/demolition_spider.py [hunk 2]
```diff
@@ -34,22 +33,57 @@ def __init__(
...
+        event_target, event_argument = m.groups()
```
Comment (thashepherd) [inline src/demolition_spider.py:70]: Nice! I'd suggest
```
m = re.search(r"__doPostBack\('(?<event_target>.+)'\s*,\s*'(?<event_argument>.*)'\)", href)
event_target = m.group('event_target')
event_argument = m.group('event_argument')
```
for clarity

File: src/demolition_spider.py [hunk 3]
```diff
@@ -34,22 +33,57 @@ def __init__(
...
+        return self.follow_postback(
+            response,
+            event_target=event_target,
+            event_argument=event_argument,
+            **kwargs,
+        )
```
Comment (thashepherd) [inline src/demolition_spider.py]: Nice! I'd suggest
```
m = re.search(r"__doPostBack\('(?<event_target>.+)'\s*,\s*'(?<event_argument>.*)'\)", href)
return self.follow_postback(
    response,
    event_target=m.group('event_target'),
    event_argument=m.group('event_argument'),
    **kwargs,
)
```
for clarity

File: src/demolition_spider.py [hunk 4]
```diff
@@ -72,18 +106,26 @@ def parseResults(self, response: Response):
...
+        # if there are more pages, follow the next page link
+        next_button = response.xpath(
+            "//td[@class='aca_pagination_td aca_pagination_PrevNext']/a[contains(text(), 'Next')]"
+        )
+        if next_button:
+            yield self.follow_postback_link(
+                response,
+                href=next_button.xpath("@href").get(),
+                callback=self.parseResults,
+            )
```
Comment (thashepherd) [inline src/demolition_spider.py:130]: Clean

File: src/demolition_spider.py (hunk 3 above)
Comment (aldrio) [reply src/demolition_spider.py]: Cool, swapped it

**Pull request merged**